src/ops.c

```diff
@@ -260,7 +260,12 @@ shift_line(
 		count = 0;
 	}
 	else
-	    count += sw_val * amount;
+	{
+	    if ((long)sw_val * (long)amount > INT_MAX - count)
```

Comment:
`(long)sw_val * (long)amount` may overflow when `long` is 32 bits.

Reply:
I guess that's the reason why the test fails on Windows and on i386?

Reply:
should be fixed now